tests/base/src/test/java/org/wildfly/security/auth/client/CredentialStoreSaslAuthenticationTest.java

```diff
@@ -111,4 +113,31 @@ public void testSuccessfulSaslAuthenticationWithCredentialStore() throws Excepti
 
     }
 
+    @Test
+    public void testUnsuccessfulSaslAuthenticationWithCredentialStore() throws Exception {
+        SaslServer server = new SaslServerBuilder(PlainSaslServerFactory.class, PLAIN)
+                .setProviderSupplier(() -> providers)
+                .setUserName(USERNAME)
+                .setPassword((USERNAME + PASSWORD).toCharArray())
+                .build();
+
+        // Create SASL client from XML configuration file
+        // XML configuration file specifies PASSWORD instead of USERNAME + PASSWORD
+        // Server configuration is specified as USERNAME + PASSWORD to create a mismatch
+        AuthenticationContext context = AuthenticationContext.getContextManager().get();
+
+        AuthenticationContextConfigurationClient contextConfigurationClient = AccessController.doPrivileged(AuthenticationContextConfigurationClient.ACTION);
+        SaslClient client = contextConfigurationClient.createSaslClient(new URI(CREDENTIAL_CONFIG_FILE), context.authRules.getConfiguration(), Arrays.asList(new String[]{PLAIN}));
+
+        assertTrue(client.hasInitialResponse());
+        byte[] message = client.evaluateChallenge(new byte[0]);
+        assertEquals("\0"+USERNAME+"\0" + PASSWORD,new String(message, StandardCharsets.UTF_8));
+
+        try {
+            server.evaluateResponse(message);
+            fail("Expected exception but no exception thrown");
+        } catch (Exception e) {
+            assertTrue(e instanceof SaslException); // expected a SASL exception due to failed authentication
```

Comment:
Instead of adding the assertion here, we could catch the expected exception directly:

```
} catch (AuthenticationMechanismException e) {
     // ignored
}
```

Reply:
Actually we should catch `SaslException` directly since that is what `evaluateResponse` throws:

```
} catch (SaslException e) {
     // ignored
}
```

Reply:
That's great! 

It doesn't look like I see the update here yet. It stills shows for me like this:

```
} catch (Exception e) {
      assertTrue(e instanceof SaslException); // expected a SASL exception due to failed authentication
}
```

instead of just

```
} catch (SaslException e) {
     // ignored
}
```